### DISCUSSION TITLE
AI And Ballast Flora - Suggestions

### DISCUSSION THREAD

**Discussion by BriannaTheBrave** [FakeFishGames/Barotrauma#4831] [Ideas and suggestions]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
I play the campaign by myself. 

It is harder to keep the AI bots from dying to my efforts to fight a Ballast Flora than the risk of dying from Ballast Flora.

To make it easier to move the AI out of the way so I can fight the Ballast Flora, I suggest the following:

There needs to be a "Every AI follow me" to move them all a

There would be an "Every AI Wait here" 

I think under emergency there should be a "Stop All" option which prevents the AI from doing anything (fire fighting, repairs, welding) other than equipping diving masks and shooting/meleeing enemies who come near. This should persist until I cancel this mode. I have killed many crew AI with my flamer as they run through it to put out fires or try to fix hull damage.

**Steps To Reproduce**
You have been infected by Ballast Flora (in a vanilla sub like a humpback or typhoon)

In this scenario the AI bots will continue to go about their business, and need to be cleared away as a flamer and explosives are needed. Often the AI would just run into my jet stream of fire to try and put out the fire or fix a damaged hull. It takes a lot of time to order each one to follow me, then to stay put, and only then can I go to fight the Ballast Flora.

They also will not stay away from the ballast flora until their own health is very low. They do not run away like a normal intruder, even crawler.

Even if we take the approach to re-wire the ballast and use a plasma cutter method, the AI will not avoid the ballast emissions. I need to move many bots and reassign each one individually, first to come to a safe location, then to stay there. This has made the game very frustrating as the Ballast flora is very interesting and insidious, but sadly my campaign has to stop until something is done to give me better control.

**Version**
Version 10 (Steam) - Windows 10

**Additional information**
Note that ballast flora is cool! It is a unique threat and I like it, however because it is hard to keep the AI away from it this makes the game very frustrating to play by myself. 

Also the fact that flame throwers cannot strike through an open floor hatch is a problem for sweeping ballast flora between levels.

Also the steam cannon (legendary flamer thrower) does not do damage to the ballast flora and this seems odd to me. I know you did not want to be easy to kill and the ballast is smart and uses water to protect itself but this tool seemed like a perfect fit and was very disappointing.

**Comment by Entelin** [comment]
No need to stop your campaign over this. It is not nessesary to "sweep" ballast flora from levels of the ship. Once the source on the pump is dead, it is gone entirely. So drain the ballast tank by pinning the ship against the ceiling, you can use hold position to do this by selecting a position inside the wall, or if you are playing the captain you can set manual controls to full up. Grab a suit and medical supplies, then go into the affected ballast tank, turn off the tank, and start using the plasma cutter until it's gone. I agree with some of your points on the ai though. Personally I failed 4-5 times to kill the flora before using the above solution. I tried 2 C-4 detonators, but not only did it not kill flora, it also nearly totally wrecked my small ship and killed ai that were nearby (so an order for all to move would be great for this). The flamer (predictably) just made fires which were themselves rather hard to survive in a small room. So I see the plasma cutter really the only viable method currently, even in multiplayer when people could move away from a flamer there's no point in using it as they could help you plasma cut it instead. Far better to let the ai continue repairing the ship, manning the guns, etc. Rewiring the ballast tank to have a manual empty switch that bypasses the navcon is also easy to do in game and helps speed up the process of removing it since you can skip the "pin yourself against the ceiling" step and just hit the switch and get to work. The visual indicator that you are damaging the flora could be more clear however, it's really non obvious that anything is happening to it unless you know what you are looking for.

**Reply by mLuby** [reply]
Just wanted to add that this isn't limited to bots. I was captaining a campaign with a crew of randoms who—though otherwise competent—didn't have much Ballast Flora experience. As a result, the human players unintentionally hindered efforts to eradicate the infestation, by ineffectually using plasma cutters while poisoning the air and getting too close to the purifying flames.

I'll add that telling bots to "follow" and "stay put" is helpful to move them about.

**Comment by Sviat** [comment]
Assigning bots to Guns makes them safe for some time...

**Comment by RadCatish** [comment]
Could we get an "All players follow me" and "All players wait here" too? 🤣 We've lost too many people to fighting ballast flora

**Reply by Sviat** [reply]
these could be assigned on general (in contrast to personal) bot command
